libpromises/verify_classes.c

```diff
@@ -147,20 +147,34 @@ static bool SelectClass(EvalContext *ctx, const Rlist *list, const Promise *pp)
         PromiseRef(LOG_LEVEL_ERR, pp);
         return false;
     }
-    assert(list);
+    else if (count == 1) /* There is no point of computation if there is only 
+                          * one item on a list. */
+    {
+        if (IsVarList(RlistScalarValue(list)))
+        {
+            Log(LOG_LEVEL_VERBOSE,
+                "select_class: Can not expand list '%s' for setting class.",
```

Comment:
So will this result in a message that indicates the promise is skipped because the list cannot be expanded? (I don't know if this will trigger some other log message that specifically says skipped). I think we should call it out specifically.